### ISSUE TITLE
Ability to group components

### ISSUE THREAD

**Issue by iwootten** [measuredco/puck#77]
When navigating a large collection of components, it would be useful to have them grouped into some form of hierarchy (rather than a single list) so they could be found more easily.

**Comment by xaviemirmon** [comment]
@iwootten I'd also find this super useful (if not necessary) for my use-case. @chrisvxd wouldn't it be just a matter of adding an option group label to the component config?  From there we could group into an accordion in the sidebar.  Or, is there some extra complexity I'm missing?

**Comment by chrisvxd** [comment]
+1 for tagging @ahmedrowaihi. This will be additionally useful for restricting what can be dropped into dropzones once #37 lands.

I'll spin out a separate ticket for that one.

**Comment by chrisvxd** [comment]
I think I was slightly confused  around the initial proposal here; I thought it was to do with grouping components that have been dragged into the page, rather than  in the list of available components, but believe I misinterpreted it.

Regardless, I think #87 should cover this ticket now, and includes some proposals.

I'll spin out yet another ticket for what I presumed this one was about.

If nobody shouts otherwise, I'll close this ticket in favour of #87.

Sorry for the confusion!

**Comment by chrisvxd** [comment]
The "other" type of grouping: https://github.com/measuredco/puck/issues/88

**Comment by chrisvxd** [comment]
Nobody shouted, so closing this ticket in favour of #87 and #88, which are more explicit and include proposals

**Comment by iwootten** [comment]
Been at a conference the last couple of days so haven't been able to reply to this. Apologies.

> I think I was slightly confused around the initial proposal here; I thought it was to do with grouping components that have been dragged into the page, rather than in the list of available components, but believe I misinterpreted it.
>

Correct, grouped in the left hand side (prior to placement) is what I meant.

> Regardless, I think #87 should cover this ticket now, and includes some proposals.

Sounds good!
